### PR TITLE
fix to apply command line arguments in k4run again

### PR DESCRIPTION
**Pull request by andrea1794**

### CODE REVIEW

**Comment by vvolkl** [comment]
Hi Andrea,

Thanks for this patch! Could you add a comment explaining why exactly these lines are needed, what they are supposed to do, and what issue they fix for DIRAC? 

It also looks to me like this results in a rather long printout on every invocation of k4run. Just running pythia8 alone, I already get
```
~/repo/k4fwcore/k4FWCore/scripts ((6f24aa0...))$ ./k4run ~/repo/fccsw/Examples/o
ptions/pythia.py 
# WARNING: Found 1 duplicates among the 10 genConfDb files :
# WARNING: --------------------------------------------------
# WARNING:   -<component name>: <module> - [ <duplicates> ]
# WARNING: --------------------------------------------------
# WARNING:   -CreateExampleEventData: k4ProjectTemplate.k4ProjectTemplatePluginsConf - ['k4TestFWCore.k4TestFWCorePluginsConf']
# WARNING: Fix your cmt/requirements file !!
 -->  Pythia8 -->  HepMCToEDMConverter -->  StableParticles -->  out 

Option name: Pythia8.OutputLevel Pythia8.OutputLevel 0
Option name: out.OutputLevel out.OutputLevel 0
Option name: out.filename out.filename output.root
Option name: out.outputCommands out.outputCommands ['keep *']
Option name: out.remoteFilename out.remoteFilename 
Option name: Pythia8.PythiaInterface.OutputLevel Pythia8.PythiaInterface.OutputLevel 0
Option name: Pythia8.PythiaInterface.Filename Pythia8.PythiaInterface.Filename /cvmfs/sw.hsf.org/spackages2/k4gen/0.1pre02/x86_64-centos7-gcc8.3.0-opt/nyi7pl2fn5zuj65majyzpffuu6pokt6i/share/k4Gen/data/Pythia_standard.cmd
Option name: Pythia8.PythiaInterface.printPythiaStatistics Pythia8.PythiaInterface.printPythiaStatistics True
Option name: Pythia8.PythiaInterface.doEvtGenDecays Pythia8.PythiaInterface.doEvtGenDecays False
Option name: Pythia8.PythiaInterface.EvtGenDecayFile Pythia8.PythiaInterface.EvtGenDecayFile Generation/data/DECAY.DEC
Option name: Pythia8.PythiaInterface.UserDecayFile Pythia8.PythiaInterface.UserDecayFile 
Option name: Pythia8.PythiaInterface.EvtGenParticleDataFile Pythia8.PythiaInterface.EvtGenParticleDataFile Generation/data/evt.pdl
Option name: Pythia8.GaussSmearVertex.OutputLevel Pythia8.GaussSmearVertex.OutputLevel 0
Option name: Pythia8.GaussSmearVertex.xVertexSigma Pythia8.GaussSmearVertex.xVertexSigma 0.5
Option name: Pythia8.GaussSmearVertex.yVertexSigma Pythia8.GaussSmearVertex.yVertexSigma 0.5
Option name: Pythia8.GaussSmearVertex.zVertexSigma Pythia8.GaussSmearVertex.zVertexSigma 40.0
Option name: Pythia8.GaussSmearVertex.tVertexSigma Pythia8.GaussSmearVertex.tVertexSigma 0.18
Option name: Pythia8.GaussSmearVertex.xVertexMean Pythia8.GaussSmearVertex.xVertexMean 0.0
Option name: Pythia8.GaussSmearVertex.yVertexMean Pythia8.GaussSmearVertex.yVertexMean 0.0
Option name: Pythia8.GaussSmearVertex.zVertexMean Pythia8.GaussSmearVertex.zVertexMean 0.0
Option name: Pythia8.GaussSmearVertex.tVertexMean Pythia8.GaussSmearVertex.tVertexMean 0.0
Option name: EventDataSvc.OutputLevel EventDataSvc.OutputLevel 0
Option name: EventDataSvc.input EventDataSvc.input 
Option name: HepMCToEDMConverter.OutputLevel HepMCToEDMConverter.OutputLevel 0
Option name: StableParticles.OutputLevel StableParticles.OutputLevel 0
Option name: StableParticles.accept StableParticles.accept [1]
ApplicationMgr    SUCCESS 
====================================================================================================================================
```


 The output of Gaudi jobs can already be a bit cluttered, so I'd like to avoid adding these printouts. The output of `k4run <optionsfile> -h` does not let you do what you are trying to do?

**Comment by andresailer** [comment]
This is needed to actually apply the command line arguments to the properties (not just for DIRAC), it got lost in https://github.com/key4hep/k4FWCore/pull/39/files#diff-bf40c8105fb694b5f17d061ec4fd472791c8907ad3358653775956d61e249237L135-L138

The `print`s should maybe be changed to use logger and this print here could be done at debug level (but for another PR).

**Comment by andrea1794** [comment]
Hi Valentin,
sorry for the bad commit message...

And thanks Andre for the clarification!
Should I change the print line then?

**Comment by vvolkl** [comment]
Good catch! The lost lines should have been noticed in some test. I added https://github.com/key4hep/k4FWCore/pull/49 for the future.
Yes,  please remove the print line, then I'll merge.

File: k4FWCore/scripts/k4run [hunk 1]
```diff
@@ -127,6 +127,11 @@ if __name__ == "__main__":
       print("usage: " + os.path.basename(__file__) + " [gaudi_config.py ...] Specify a gaudi options file to run")
       sys.exit()
 
+    opts_dict = vars(opts)
+    for optionName, propTuple in option_db.items():
+      print("Option name:", propTuple[1], optionName, opts_dict[optionName])
```
Comment (andresailer) [inline k4FWCore/scripts/k4run:132]: When https://github.com/key4hep/k4FWCore/issues/50  is done, add this as `debug` output

**Comment by vvolkl** [comment]
ok, I'll merge now and add the logging before the next tag

**Pull request merged**